modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -3980,11 +3980,18 @@ all. It returns the rotated rectangle in which the ellipse is inscribed. The fir
 is used. Developer should keep in mind that it is possible that the returned
 ellipse/rotatedRect data contains negative indices, due to the data points being close to the
 border of the containing Mat element.
+If the flag direct is true a direct ellipse fit algorithm is used decribed by @cite Fitzgibbon99
 
 @param points Input 2D point set, stored in std::vector\<\> or Mat
  */
 CV_EXPORTS_W RotatedRect fitEllipse( InputArray points );
 
+/** @overload
+@param points Input 2D point set, stored in std::vector\<\> or Mat
+@param direct Flag to use a direct ellipse fit algorithm
+*/
+CV_EXPORTS_AS(fitEllipse2) RotatedRect fitEllipse( InputArray points, bool direct );
```

Comment:
Thank you for your review. I will change this within the next commit.
I also tested the fit with the given test case. Compared to old implementation, #4515 works fine whereas #6544 still returns wrong ellipse. I will check this perhaps I can find the issue in the new algorithm.

I have seen the paper from the bug report, linked [here](http://autotrace.sourceforge.net/WSCG98.pdf). There is some Matlab code, so I think it is not that complicated to rewrite this to OpenCV code.
I also would change the argument `bool direct` into `int method` to get more flexibility in choosing algorithms. Give me a few days to check this.